File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015CS/VirtualRadGridViewSortingFiltering.Core/VirtualGridViewElement.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class VirtualRadGridViewElement : RadGridViewElement
    {
        public ItemSource ItemsSource
        {
            get { return (this.Template.ListSource as VirtualGridViewListSource).DataView.ItemsSource; }
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadGridViewElement);
            }
        }

        protected override MasterGridViewTemplate CreateTemplate()
        {
            return new VirtualMasterGridViewTemplate();
        }
    }
}

File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015VB/VirtualRadGridViewSortingFiltering.Core/VirtualGridViewElement.vb

Imports System.Collections.Generic
Imports System.Text
Imports Telerik.WinControls.UI

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class VirtualRadGridViewElement
        Inherits RadGridViewElement
        Public ReadOnly Property ItemsSource() As ItemSource
            Get
                Return TryCast(Me.Template.ListSource, VirtualGridViewListSource).DataView.ItemsSource
            End Get
        End Property

        Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
            Get
                Return GetType(RadGridViewElement)
            End Get
        End Property

        Protected Overrides Function CreateTemplate() As MasterGridViewTemplate
            Return New VirtualMasterGridViewTemplate()
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
